control-plane/connect-inject/peering_acceptor_controller.go

```
@@ -0,0 +1,335 @@
+	if peeringAcceptor.Status.Secret.Backend != peeringAcceptor.Spec.Peer.Secret.Backend {
+		return false, false, errors.New("PeeringAcceptor backend cannot be changed")
+	}
+	// Compare the existing secret hash.
+	// Get the secret specified by the status, make sure it matches the status' secret.latestHash.
+	if existingSecret != nil {
+		existingSecretHashBytes := sha256.Sum256(existingSecret.Data[peeringAcceptor.Status.Secret.Key])
+		existingSecretHash := hex.EncodeToString(existingSecretHashBytes[:])
+		if existingSecretHash != peeringAcceptor.Status.Secret.LatestHash {
+			return true, false, nil
+		}
+
+	}
+	return false, false, nil
+}
+
+// updateStatusError updates the peeringAcceptor's secret in the status.
+func (r *PeeringAcceptorController) updateStatus(ctx context.Context, peeringAcceptor *consulv1alpha1.PeeringAcceptor, resp *api.PeeringGenerateTokenResponse) error {
+	peeringAcceptor.Status.Secret = &consulv1alpha1.SecretStatus{
+		Name:    peeringAcceptor.Spec.Peer.Secret.Name,
+		Key:     peeringAcceptor.Spec.Peer.Secret.Key,
+		Backend: peeringAcceptor.Spec.Peer.Secret.Backend,
+	}
+
+	peeringTokenHash := sha256.Sum256([]byte(resp.PeeringToken))
+	peeringAcceptor.Status.Secret.LatestHash = hex.EncodeToString(peeringTokenHash[:])
+
+	peeringAcceptor.Status.LastReconcileTime = &metav1.Time{Time: time.Now()}
+	peeringAcceptor.Status.ReconcileError = &consulv1alpha1.ReconcileErrorStatus{
+		Error:   pointerToBool2(false),
+		Message: pointerToString(""),
+	}
+	err := r.Status().Update(ctx, peeringAcceptor)
+	if err != nil {
+		r.Log.Error(err, "failed to update PeeringAcceptor status", "name", peeringAcceptor.Name, "namespace", peeringAcceptor.Namespace)
+	}
+	return err
+}
+
+// updateStatusError updates the peeringAcceptor's ReconcileError in the status.
+func (r *PeeringAcceptorController) updateStatusError(ctx context.Context, peeringAcceptor *consulv1alpha1.PeeringAcceptor, reconcileErr error) error {
```

We always ignore the error that's returned by this function. I'm assuming we shouldn't, so maybe it could be a TODO?

---

I don't think there's anything we can do to handle the error from this function. If we fail to update the status with the error message, we don't want the status update failure to be what's returned from the reconcile, instead we want the real error. I suppose we could have this function just not return an error, and document that we are ignoring any errors in this case? The only error case is if the K8s api is down when trying to make an update to the status, and instead the real error would be returned as a reconcile error, wdyt?